cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -122,6 +133,24 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     String subnet = conf.getSubnet();
     Network networkInfo = getNetworkInfo(projectId, network, compute);
 
+    boolean useInternalIP;
+    
+    // if a peered network is specified then the instance is tied to a network
+    PeeringState state = peeringState(systemNetwork, systemProjectId, networkInfo);
+    if (state == PeeringState.ACTIVE) {
+      if (conf.isPreferExternalIP()) {
+        LOG.warn("Prefer external IP setting will be ignored and internal IP will used to communicate with the " +
```

Comment:
We should not be making an assumption that the user has misconfigured this property. If a user is explicitly telling the provisioner to use the external IP, it should always use the external IP, regardless of how the network is setup. We can log an INFO message that the network is peered and suggest that the internal IP be used, but we should not override something they have explicitly set.

Reply:
fixed.